src/main/java/Duke.java

```diff
@@ -1,10 +1,236 @@
+     * @param isMark boolean true or false to check if the task has been marked.
+     */
+    public static void printMarkOrUnmarkTask(int markIndex, boolean isMark) {
+        if (isMark) {
+            System.out.println("\t Awesome! I've marked this task as done:");
+        } else {
+            System.out.println("\t Awesome! I've marked this task as not done yet:");
+        }
+        System.out.println("\t\t [" + assignments[markIndex].getStatusOfTypeTask()
+                + "]" + "[" + assignments[markIndex].getStatusOfDone()
+                + "] " + assignments[markIndex].description);
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+        String lineDivider = printWelcomeMessage();
+        String userInput;
+        Scanner in = new Scanner(System.in);
+        boolean isBye;
+
+        do {
+            //Enable user to enter text
+            userInput = in.nextLine();
+            System.out.print("\t" + lineDivider);
+            String[] splitUserInputs = userInput.split(" ", 2);
+            isBye = splitUserInputs[0].equals("bye");
+
+            boolean isUserInputData = !splitUserInputs[0].equals("bye")
+                    && !splitUserInputs[0].equals("list")
+                    && !splitUserInputs[0].equals("mark")
+                    && !splitUserInputs[0].equals("unmark");
+            boolean isList = splitUserInputs[0].equals("list");
+            boolean isMark = splitUserInputs[0].equals("mark");
+            boolean isUnmark = splitUserInputs[0].equals("unmark");
+
+            if (isUserInputData) {
+                sortTypeOfTask(userInput, splitUserInputs);
+            } else if (isList) {
+                printList();
+            } else if (isMark) {
+                int indexMark = Integer.parseInt(splitUserInputs[1]) - 1;
+                //To handle a case where user tries to mark a task that has not been specified
+                if (hasMarkTaskError(indexMark, lineDivider)) {
+                    continue;
+                }
+                printMarkOrUnmarkTask(indexMark, true);
+            } else if (isUnmark) {
+                int indexMark = Integer.parseInt(splitUserInputs[1]) - 1;
+                //To handle a case where user tries to unmark a task that has not been specified
+                if (hasUnmarkTaskError(indexMark, lineDivider)){
+                    continue;
+                }
+                printMarkOrUnmarkTask(indexMark, false);
+            } else {
+                System.out.println("\t Bye. Hope to see you again soon!");
+            }
+            System.out.println("\t" + lineDivider);
+        } while(!isBye);
```

Comment:
This code violates the coding standard. Consider adding spaces between `while` and `(` to comply with coding standards.